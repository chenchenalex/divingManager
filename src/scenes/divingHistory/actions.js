import { USER_FETCH_DATA_ASYNC } from "src/actions/actionTypes";

const userFetchDataAsync = userId => {
  if (typeof userId === "undefined")
    throw new Error("User Id is mandatory to fetching dive data");

  return {
    type: USER_FETCH_DATA_ASYNC,
    payload: {
      userId,
      section: ""
    }
  };
};

export default userFetchDataAsync;
