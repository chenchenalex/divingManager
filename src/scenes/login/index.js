import account from "./login";

export default account;
