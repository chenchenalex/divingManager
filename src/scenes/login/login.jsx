import React from "react";
import "firebase/auth";
import { LOGIN_FORM_CONFIG } from "src/data/config";
import Notification from "src/components/notification";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { ACCOUNT } from "src/data/routes";
import { userLogin } from "src/services/firebase";
import PropTypes from "prop-types";

import LoginContainer from "./style";
import LoginForm from "./components/loginForm";

export class LoginPage extends React.Component {
  state = {
    username: "",
    password: "",
    errorMessage: ""
  };

  onFormSubmit = e => {
    e.preventDefault();

    const { username, password } = this.state;

    this.setState({
      errorMessage: "",
      loading: true
    });

    return userLogin({ username, password })
      .then(() => {
        this.setState({
          loading: false
        });
      })
      .catch(({ message }) => {
        this.setState({
          errorMessage: message,
          loading: false
        });
      });
  };

  onChange = e => {
    const { name } = e.target;

    this.setState({
      [name]: e.target.value
    });
  };

  formConfig = JSON.parse(JSON.stringify(LOGIN_FORM_CONFIG));

  render() {
    let redirectUrl;
    if (typeof this.props.location.from !== "undefined") {
      redirectUrl = this.props.location.from.pathname;
    } else {
      redirectUrl = ACCOUNT;
    }

    return this.props.userInfo.isAuthenticated ? (
      <Redirect to={{ pathname: redirectUrl }} />
    ) : (
      <LoginContainer>
        <div className="content">
          <div className="notifcation-area">
            {this.state.errorMessage !== "" && (
              <Notification variant="error">
                {this.state.errorMessage}
              </Notification>
            )}
          </div>

          <h1>LOGIN</h1>

          <LoginForm
            formConfig={this.formConfig}
            onSubmit={this.onFormSubmit}
            onChange={this.onChange}
            data={this.state}
            isloading={this.state.loading}
          />

          <div className="mask" />
        </div>
      </LoginContainer>
    );
  }
}

LoginPage.propTypes = {
  location: PropTypes.object.isRequired,
  userInfo: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    userInfo: state.userInfo
  };
}

export default connect(mapStateToProps)(LoginPage);
