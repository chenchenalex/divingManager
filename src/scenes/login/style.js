import styled from "styled-components";
import bgImage from "src/assets/images/dust_scratches.png";

export const AccountContainer = styled.div`
  background: url(${bgImage});
  min-height: 100%;
  position: relative;
  overflow: hidden;

  .content {
    width: 400px;
    margin: 0 auto;
    padding-top: calc(var(--base-unit) * 3);
    color: var(--teal);
    border-radius: 5px;
    z-index: 2;
  }

  .mask {
    position: absolute;
    bottom: -150px;
    right: -150px;
    height: 300px;
    width: 300px;
    transform: rotateZ(45deg);
    z-index: 1;
    background-color: black;
    opacity: 0.5;
  }

  .notifcation-area {
    min-height: calc(var(--base-unit) * 14);
  }

  h1 {
    margin: calc(var(--base-unit) * 5) 0;
    font-size: var(--font-size-xl);
  }

  label,
  input {
    color: var(--teal);
  }

  .form-actions__register {
    margin: calc(var(--base-unit) * 2) 0;
    font-size: var(--font-size-sm);

    a {
      color: var(--teal);
      &:hover {
        text-decoration: underline;
      }
    }
  }

  button {
    margin-top: calc(var(--base-unit) * 2);
  }
`;
