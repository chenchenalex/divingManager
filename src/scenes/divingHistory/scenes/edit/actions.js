import { ADD_DIVE, EDIT_DIVE } from "src/actions/actionTypes";

export const addNewDive = formData => ({
  type: ADD_DIVE,
  payload: formData
});

export const editDive = formData => ({
  type: EDIT_DIVE,
  payload: formData
});
