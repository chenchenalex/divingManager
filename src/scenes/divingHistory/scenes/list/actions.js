import { DELETE_DIVE } from "src/actions/actionTypes";

const deleteActionCreator = (itemsToBeDeleted = []) => ({
  type: DELETE_DIVE,
  payload: itemsToBeDeleted
});

export default deleteActionCreator;
