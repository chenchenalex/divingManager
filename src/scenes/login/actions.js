import { USER_LOGIN, USER_LOGIN_SUCCESS } from "src/actions/actionTypes";

export const loginAction = (username, password) => {
  return {
    type: USER_LOGIN,
    payload: { username, password }
  };
};

export const loginSuccess = () => {
  return {
    type: USER_LOGIN_SUCCESS,
    payload: {}
  };
};
