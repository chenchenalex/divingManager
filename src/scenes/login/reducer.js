import { USER_LOGIN_SUCCESS } from "src/actions/actionTypes";

export default function login(state = {}, action) {
  switch (action.type) {
    case USER_LOGIN_SUCCESS:
      return {
        ...state,
        isAuthenticated: true
      };
    default:
      return state;
  }
}
