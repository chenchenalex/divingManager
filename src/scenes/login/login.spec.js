import { LoginPage } from "./login";
import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { shallow } from "enzyme";
import { userLogin } from "src/services/firebase";
import { loginSuccess } from "src/actions";

// Jest mock all implementation of firease
jest.mock("src/services/firebase");
jest.mock("src/actions");

beforeEach(() => {
  userLogin.mockClear();
  loginSuccess.mockClear();
});

describe("login page unit tests", () => {
  const userInfo = { isAuthenticated: false };
  const location = {
    from: "/jest"
  };
  const component = shallow(
    <LoginPage userInfo={userInfo} location={location} />
  );

  it("onSubmit(): should submit username and password", async () => {
    const instance = component.instance();

    instance.state = {
      username: "alex",
      password: "123"
    };

    userLogin.mockImplementation(() => {
      return Promise.resolve();
    });

    await instance.onFormSubmit({ preventDefault: jest.fn() });

    expect(userLogin).toHaveBeenCalledWith({
      username: "alex",
      password: "123"
    });
  });

  it("Snapshot: Login page", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <LoginPage userInfo={userInfo} location={location} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
