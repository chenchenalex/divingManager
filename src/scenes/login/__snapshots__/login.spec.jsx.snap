// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page unit tests Snapshot: Login page 1`] = `
<div
  className="sc-bwzfXH iJHrGE"
>
  <div
    className="content"
  >
    <div
      className="notifcation-area"
    />
    <h1>
      LOGIN
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
          data-shrink={false}
        >
          Email
          <span
            className="MuiFormLabel-asterisk-14"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            className="MuiInput-input-23 MuiInput-inputType-26"
            disabled={false}
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
          data-shrink={false}
        >
          Password
          <span
            className="MuiFormLabel-asterisk-14"
          >
             *
          </span>
        </label>
        <div
          className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            className="MuiInput-input-23 MuiInput-inputType-26"
            disabled={false}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="form-actions"
      >
        <p
          className="form-actions__register"
        >
          New User?  
          <a
            href="/register"
            onClick={[Function]}
          >
            <strong>
              Sign up
            </strong>
          </a>
        </p>
        <div
          className="m-top__md"
        >
          <button
            className="MuiButtonBase-root-44 MuiButton-root-28 MuiButton-raised-34 MuiButton-raisedSecondary-36"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="submit"
          >
            <span
              className="MuiButton-label-29"
            >
              Login
            </span>
            <span
              className="MuiTouchRipple-root-47"
            />
          </button>
        </div>
      </div>
    </form>
    <div
      className="mask"
    />
  </div>
</div>
`;
